### PR TITLE
DBZ-6118: Cleanup Mongo oplog code logic

### PR DESCRIPTION
**Pull request by xinbinhuang**
https://issues.redhat.com/browse/DBZ-6118

- Move two `*RecordEmitter.java` class under `record_emitter` package
- Rename `*oplog*` -> `*snapshot*` whenever possible
- Remove `oplog` specific schema fields

### CODE REVIEW

**Comment by jpechane** [comment]
@jcechace Could you please take care of this one?

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/record_emitter/MongoDbSnapshotRecordEmitter.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.record_emitter;
```
[the remark this answers is not in the document]
Comment (jpechane) [reply debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/record_emitter/MongoDbSnapshotRecordEmitter.java:6]: I'd omit the underscore

**Comment by jcechace** [comment]
@xinbinhuang Thanks for the contribution. Can you address those failing tests and rename the `record_emitter` package to `recordemitter`, please?

**Comment by xinbinhuang** [comment]
Extra fixed to update `ExtractNewDocumentState` to properly parse change stream output and tests with it (5e797c17ecee7f9ea54e180b0865322d95ff376a)
- TODO: handling `truncatedArrays`?

**Comment by xinbinhuang** [comment]
r? PTAL @jcechace

**Comment by jcechace** [comment]
@xinbinhuang since we are not breaking any tests I suggest creating a follow up Jira for the truncated arrays (please describe what seems to be the issue as I don't fully understand what the problem is). 

I will merge this once we have the followup Jira captured.

**Review by jcechace — APPROVED** [review]

**Comment by xinbinhuang** [comment]
@jcechace jira created: https://issues.redhat.com/browse/DBZ-6141

> (please describe what seems to be the issue as I don't fully understand what the problem is).

Mostly because I'm not familiar with the usage of `truncatedArrays` and would need more testing to handle it properly. Since this PR is already getting large and `truncatedArrays` is some sort of new functionality (not existed in legacy oplog mode), so doing it in another PR will be cleaner IMO

**Comment by jcechace** [comment]
Great work @xinbinhuang, thanks for this contribution!

**Pull request merged**